packages/grpc-native-core/index.d.ts

```diff
@@ -294,6 +294,8 @@ declare module "grpc" {
     handleServerStreamingCall<RequestType, ResponseType> |
     handleBidiStreamingCall<RequestType, ResponseType>;
 
+  type authContext = object | null;
```

Comment:
I don't think this is a good type for this. I don't see why it should ever be `null`, and we can be a lot more specific about the fields. It should probably be something like

```ts
interface authContext = {
  transportSecurityType: string;
  sslPeerCertificate: Certificate;
}
```

My understanding is that they will always be populated. Otherwise it should be `?:` for optional fields.

Reply:
That's fair about the "object" type. I didn't necessarily try too hard defining it.

As for the "null" portion, this goes with what the current implementation does in C++. I may have been a bit over protective, but I'm returning `null` if there's a null auth context from the core, but I'm not sure of what that could imply, to be fair.

Reply:
I don't like having too many layers of "maybe this exists, maybe it doesn't". I would prefer to make both properties optional, and have an empty object if `null` comes up from the core. That being said, does a missing `transportSecurityType` even have meaningful semantics?

Reply:
So, yeah, we could be strengthening this a bit more, I don't disagree. My issue is that the core's auth context isn't really... well... defined in terms of what things mean semantically. So on one hand I'm with you on providing a stronger upper layer by reinterpreting what the core means, but on the other, I'm wary of unknown semantic meanings that may even change in the future in a way that might surprise us in an unfortunate manner, so it's kind of a coin toss for me.

Reply:
I guess part of the question is what we are communicating to the users. The way I see it, an absent property in the object we return means "we don't have this information", which is true if it's absent from the auth context from core, or if the auth context from core is null. Do we have good reason to believe that our users will ever get any useful information from getting `null` vs `{}`? I am OK with omitting `transportSecurityType` when we don't get it from the core.